pkgs/servers/mail/public-inbox/default.nix

```diff
@@ -59,7 +70,12 @@ buildPerlPackage rec {
   installTargets = [ "install" ];
   postInstall = ''
     for prog in $out/bin/*; do
-        wrapProgram $prog --prefix PATH : ${lib.makeBinPath [ git ]}
+        wrapProgram $prog --prefix PATH : ${lib.makeBinPath [
+          git
+          /* for InlineC */
+          gnumake
+          stdenv.cc
```

Comment:
This will break if cross-compiled.  Is there no way to compile InlineC stuff up front at build time?

Reply:
Well, according to [public-inbox-tuning](https://public-inbox.org/public-inbox-tuning.html), recent `public-inbox` generates and compiles C code at runtime to increase performance:
```
  Process spawning
    Our optional use of Inline::C speeds up subprocess spawning from large
    daemon processes.

    To enable Inline::C, either set the "PERL_INLINE_DIRECTORY" environment
    variable to point to a writable directory, or create
    "~/.cache/public-inbox/inline-c" for any user(s) running public-inbox
    processes.

    More (optional) Inline::C use will be introduced in the future to lower
    memory use and improve scalability.
```
I've little experience with cross-compiling using Nixpkgs, so I don't know when or how it will break, nor how to address that concern. Should I use something else than `stdenv.cc` to get the target's compiler? Should I put `InlineC` usage behind a package option?

Reply:
Ah, I see -- Inline::C is different to XS, in that the code is generated at runtime.  So we can't compile at ahead of time and will need a C compiler.  In that case, the usual approach I've seen here (that is cross-friendly) is to wrap with `gcc` instead of `stdenv.cc.`

Reply:
Inline::C is cool and all, but...
Has anyone mentioned to upstream that the lack of support for ahead of time compilation/the dependency on a compiler at runtime, is fairly annoying for packaging? I'm sure other package managers would also prefer to not add a dependency on gcc to public-inbox, I don't think that's just us. (And the public-inbox maintainer is, I think, very responsive to such concerns)

Reply:
`Inline::C` is optional